src/common/low_precision_transformations/src/gather.cpp

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "low_precision/gather.hpp"
+
+#include <memory>
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+#include "low_precision/network_helper.hpp"
+#include "low_precision/rt_info/precision_preserved_attribute.hpp"
+#include "itt.hpp"
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+std::shared_ptr<opset1::Constant> gatherDeqConstant(
+    const std::shared_ptr<ngraph::Node> gather,
+    const std::shared_ptr<ngraph::Node> dequantizaitonConstant) {
+    auto constant = ov::as_type_ptr<ngraph::opset1::Constant>(dequantizaitonConstant);
+    auto constantShape = constant->get_shape();
+    if (shape_size(constantShape) == 1ul) {
+        return NetworkHelper::toScalar(constant);
+    }
+
+    const auto gatherShape = gather->get_input_partial_shape(0);
+    const size_t rank = gatherShape.rank().get_length();
+    if (rank != constantShape.size()) {
+        ngraph::Shape newConstantShape;
+        newConstantShape = constantShape;
+        // case when constShape without batch
+        if ((constantShape.size() > 1) &&
+            (constantShape.size() < rank)) {
+            newConstantShape.insert(newConstantShape.begin(), 1);
+        }
+        constantShape = newConstantShape;
+        constant = ngraph::opset1::Constant::create(ngraph::element::i32, { newConstantShape.size() }, newConstantShape);
+    }
+
+    const int64_t axis = ov::as_type_ptr<opset1::Constant>(gather->get_input_node_shared_ptr(2))->cast_vector<int64_t>()[0];
+    const size_t normalizedAxis = normalize_axis(gather->get_friendly_name(), axis, gather->get_input_partial_shape(0).rank());
+
+    // Dequantization channel match with gather axis
+    if (constantShape[normalizedAxis] != 1ul) {
+        const auto gather7 = ov::as_type_ptr<ngraph::opset7::Gather>(gather);
+        if (gather7) {
+            const auto output = fold<ngraph::opset7::Gather>(
+                constant,
+                gather7->input_value(1),
```

Comment:
Hi @v-Golubev May I know why we need to check that indices constant has 0D or 1D shape? Is it enough to just check indices is constant?

Reply:
I suggested it in order to avoid ND dequantization operations creation. For instance, if we had per-channels dequantization operations (shape=[1, 5]) in example №2 from the [specification ](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/Gather_8.md), after dequantization propagation we would have dequantization operation with shape [2, 3]. At the moment, such dequantization couldn't be handled in most LP transformations. Previously, similar situations sometimes led to performance regressions

From the other hand, example above is mathematically correct. Moreover, the Gather itself is executed in INT8 if we don't introduce additional limitations. So maybe the best idea is to leave the logic as is and run performance validation

Reply:
Thanks for you explanation!  I add the limitation that indices constant has 0D or 1D shape, on the one hand to avoid ND dequantization operations creation, and on the other hand, if batch_dims is none zero, we can not do  constant_folding if indices constant is not 0D or 1D.